drud/ddev
# Allow global omit_containers["ddev-router"], add project config to bind_all_interfaces, fixes #3028, fixes #3184

## The Problem/Issue/Bug:

For some very special situations, where there is already a reverse proxy provided (like gitpod) it makes sense to turn off ddev-router and let the other reverse proxy do the job.

Also, people sometimes like to have ports exposed on all network interfaces, and `bind_all_interfaces: true` allows this. In most contexts it's a security problem, but maybe not if you want to share on your local network with colleagues, etc.

* #3028 
* #3184 

## How this PR Solves The Problem:

Allow global `omit_containers["ddev-router"]`
Allow project config `bind_all_interfaces: true`
Allow `host_phpmyadmin_port: 8036`
Allow `host_mailhog_port: 8025`

## Manual Testing Instructions:

`ddev config global --omit-containers=ddev-router,ddev-ssh-agent,dba` and see how it works out.

Just use the gitpod link here and experiment on gitpod.

On gitpod, you'll want to do these things:

```yaml
bind_on_all_interfaces: true
host_webserver_port: 8080
# Will ignore the direct-bind https port, which will land on 2222
host_https_port: 2222
# phpmyadmin won't be active unless the dba container is enabled
host_phpmyadmin_port: 8036
# I don't know that this would be useful on gitpod, but allows local db clients to run.
host_db_port: 3306
# Probably want to ignore these two ports
host_mailhog_port:  8025
```

```
ddev poweroff
ddev config global --omit-containers=ddev-router,ddev-ssh-agent,dba
ddev config --project-type=drupal9 --docroot=web --create-docroot
ddev composer create drupal/recommended-project -y
```

## Automated Testing Overview:

This is a very unusual usage, and I didn't add any tests.

## Related Issue Link(s):

## Release/Deployment notes:




<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/3191"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
